cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -223,15 +223,25 @@ private static void writeSubGeographicalRegion(String subGeographicalRegionId, S
     }
 
     private static void writeVoltageLevels(Network network, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context, Set<Double> exportedBaseVoltagesByNominalV) throws XMLStreamException {
+        String fictSubstationId = null;
         for (VoltageLevel voltageLevel : network.getVoltageLevels()) {
             double nominalV = voltageLevel.getNominalV();
             BaseVoltageMapping.BaseVoltageSource baseVoltage = context.getBaseVoltageByNominalVoltage(nominalV);
             if (!exportedBaseVoltagesByNominalV.contains(nominalV) && baseVoltage.getSource().equals(Source.IGM)) {
                 BaseVoltageEq.write(baseVoltage.getId(), nominalV, cimNamespace, writer, context);
                 exportedBaseVoltagesByNominalV.add(nominalV);
             }
+            Optional<String> substationId = voltageLevel.getSubstation().map(s -> context.getNamingStrategy().getCgmesId(s));
```

Comment:
Have you considered reusing the existing method `writeFictitiousSubstationFor(...)`. It could be called with the voltage level. It would create a different pair of sub/region for every voltage level, but we would not duplicate code ...

Reply:
Good catch! I will do some refactoring in this sense